pkg/networkservice/chains/nsmgr/suite_test.go

```diff
@@ -498,6 +497,204 @@ func (s *nsmgrSuite) Test_PassThroughLocalUsecase() {
 	// Path length to first endpoint is 5
 	// Path length from NSE client to other local endpoint is 5
 	require.Equal(t, 5*(nsesCount-1)+5, len(conn.Path.PathSegments))
+	for i := 0; i < len(nsReg.Matches); i++ {
+		require.Contains(t, nsesReg[i].Name, conn.Path.PathSegments[(i+1)*5-1].Name)
+	}
+
+	// Refresh
+	conn, err = nsc.Request(ctx, request)
+	require.NoError(t, err)
+
+	// Close
+	_, err = nsc.Close(ctx, conn)
+	require.NoError(t, err)
+
+	// Endpoint unregister
+	for i := 0; i < len(nsesReg); i++ {
+		_, err = s.domain.Nodes[0].EndpointRegistryClient.Unregister(ctx, nsesReg[i])
+		require.NoError(t, err)
+	}
+}
+
+//nolint:funlen
+func (s *nsmgrSuite) Test_PassThroughLocalUsecaseTree() {
+	t := s.T()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	labels := []string{"aa", "ba", "bb", "c", "d"}
+	firstExpectedPath := []string{"ba", "bb", "c", "d"}
+	secondExpectedPath := []string{"aa", "ab", "c", "d"}
+
+	nsReg := treeNS()
+	nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, nsReg)
+	require.NoError(t, err)
+
+	var nsesReg []*registry.NetworkServiceEndpoint
+	for i := 0; i < len(labels); i++ {
+		nseReg := &registry.NetworkServiceEndpoint{
```

Comment:
Please create a separate method `newPassthroughEndpoint` for this:
```go
nseReg := &registry.NetworkServiceEndpoint{
	Name:                fmt.Sprintf("endpoint-%v", labels[i]),
	NetworkServiceNames: []string{nsReg.Name},
	NetworkServiceLabels: map[string]*registry.NetworkServiceLabels{
		nsReg.Name: {
			Labels: map[string]string{
				step: labels[i],
			},
		},
	},
}

additionalFunctionality := additionalFunctionalityChain(ctx, s.domain.Nodes[0].NSMgr.URL,
	fmt.Sprintf("endpoint-client-%v", labels[i]), nseReg.NetworkServiceLabels[nsReg.Name].Labels, i == len(labels)-1)

_, err = s.domain.Nodes[0].NewEndpoint(ctx, nseReg, sandbox.GenerateTestToken, additionalFunctionality...)
require.NoError(t, err)
```

Reply:
done